### ISSUE TITLE
Attempting to use roaring from worker_thread throws "Module did not self register"

### ISSUE THREAD

**Issue by steveliles** [SalvatorePreviti/roaring-node#47]
I'm trying to spin up worker threads (https://nodejs.org/api/worker_threads.html) to take advantage of multiple cpu cores in a project that makes heavy use of roaring, but I get this error:

```
Error: Module did not self-register: '/Users/Steve/myproject/node_modules/roaring/build/Release/roaring.node'.
```

Apparently this is to do with native add-ons having to be aware that they might be used from worker threads, and for many modules the fix is relatively simple: https://github.com/nodejs/node/issues/21783#issuecomment-429637117

Documentation for context-aware addons and some worker_thread specific stuff about cleaning up resources here:
https://github.com/nodejs/node/blob/main/doc/api/addons.md#context-aware-addons 

I will try to fork and PR but I have not done any native before, so I'm not massively confident of success ...

**Comment by SalvatorePreviti** [comment]
Thank you for reporting this, yes, I never tried this in a worker thread nor wrote test for it. I am going to write some test code and can work on a fix somewhere in the next 1, 2 weeks

**Comment by SalvatorePreviti** [comment]
Ok yes is a bit more complicated than just setting the module to be context aware, I will need to fix some code that would not work in multiple isolates but is doable

**Comment by SalvatorePreviti** [comment]
Released v1.5.2 that fixes the module initialization to be compatible with worker threads and make the code work with multiple isolates - worker threads are going to work only if node is >= 13 (node 14 is the LTS version)

Please give it a go and let me know.

**Comment by steveliles** [comment]
Thanks so much! Will try soon ...
